src/commands/deploy.js

```diff
@@ -16,20 +16,263 @@ const SitesCreateCommand = require('./sites/create')
 const LinkCommand = require('./link')
 const { NETLIFYDEV, NETLIFYDEVLOG, NETLIFYDEVERR } = require('../utils/logo')
 
+const triggerDeploy = async ({ api, siteId, siteData, log, error }) => {
+  try {
+    const siteBuild = await api.createSiteBuild({ siteId })
+    log(
+      `${NETLIFYDEV} A new deployment was triggered successfully. Visit https://app.netlify.com/sites/${siteData.name}/deploys/${siteBuild.deploy_id} to see the logs.`
+    )
+  } catch (err) {
+    if (err.status === 404) {
+      error('Site not found. Please rerun "netlify link" and make sure that your site has CI configured.')
+    } else {
+      error(err.message)
+    }
+  }
+}
+
+const getDeployFolder = async ({ flags, config, site, siteData, log }) => {
+  let deployFolder
+  if (flags['dir']) {
+    deployFolder = path.resolve(process.cwd(), flags['dir'])
+  } else if (get(config, 'build.publish')) {
+    deployFolder = path.resolve(site.root, get(config, 'build.publish'))
+  } else if (get(siteData, 'build_settings.dir')) {
+    deployFolder = path.resolve(site.root, get(siteData, 'build_settings.dir'))
+  }
+
+  if (!deployFolder) {
+    log('Please provide a publish directory (e.g. "public" or "dist" or "."):')
+    log(process.cwd())
+    const { promptPath } = await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'promptPath',
+        message: 'Publish directory',
+        default: '.',
+        filter: input => path.resolve(process.cwd(), input),
+      },
+    ])
+    deployFolder = promptPath
+  }
+
+  return deployFolder
+}
+
+const validateDeployFolder = async ({ deployFolder, error }) => {
```

Comment:
Old version was:
https://github.com/netlify/cli/blob/39da8081eaca6e2c8ba58aa782541199e0df48fc/src/commands/deploy.js#L463

Which was broken in multiple places:
1. Using `e.status` instead of `e.code` (fixed in the new version)
https://github.com/netlify/cli/blob/39da8081eaca6e2c8ba58aa782541199e0df48fc/src/commands/deploy.js#L468
2. Referencing `stat.isDirectory` instead of `stat.isDirectory()` https://github.com/netlify/cli/blob/39da8081eaca6e2c8ba58aa782541199e0df48fc/src/commands/deploy.js#L480

Reply:
:man_facepalming: